### PR TITLE
GPX/TCX file formats option for File and Dropbox Synchronizers

### PR DESCRIPTION
**Pull request by opoto**
Second try to address #874
This PR adds a configuration to File and Dropbox synchronizers to export in GPX or TCX formats :
Screenshot_1579541737
Screenshot_1579541752
Other synchronizers don't show these options:
Screenshot_1579541936

Today this option is only possible for File, and only when activating the File synchronizer. This PR makes it configurable at any time, by opening the synchronizer screen.

Duplicated FORMAT field present in File's AUTH_CONFIG is removed and DB from previous version is migrated.

### CODE REVIEW

File: app/src/main/org/runnerup/db/DBHelper.java [hunk 1]
```diff
@@ -487,7 +505,7 @@ private static void insertAccount(SQLiteDatabase arg0, String name, int enabled,
             arg1.put(DB.ACCOUNT.FLAGS, flags);
         }
         //DBVERSION update, must provide dummy data
-        arg1.put(DB.ACCOUNT.FORMAT, "tcx");
+        arg1.put(DB.ACCOUNT.FORMAT, FileFormats.GPX.getValue());
```
[the remark this answers is not in the document]
Comment (opoto) [reply app/src/main/org/runnerup/db/DBHelper.java]: Done

Comment (gerhardol) [inline app/src/main/org/runnerup/db/DBHelper.java]: Why not DEFAULT_FORMAT?

File: app/src/main/org/runnerup/export/DropboxSynchronizer.java [hunk 2]
```diff
@@ -225,96 +299,50 @@ public Status upload(SQLiteDatabase db, final long mID) {
...
+            StringWriter writer = new StringWriter();
+            if (mFormat.contains(FileFormats.TCX)) {
+                TCX tcx = new TCX(db);
+                tcx.export(mID, writer);
+                s = uploadFile(db, mID, sport, writer, FileFormats.TCX.getValue());
+            }
+            if (s == Status.OK && mFormat.contains(FileFormats.GPX)) {
+                GPX gpx = new GPX(db, true, true, simplifier);
+                gpx.export(mID, writer);
+                s = uploadFile(db, mID, sport, writer, FileFormats.GPX.getValue());
```
Comment (gerhardol) [inline app/src/main/org/runnerup/export/DropboxSynchronizer.java]: To upload two files is a new, so just raising a question: Should RU try uploading the second also if the first fails?
(Ignoring how FileSynch works, that isa litlle broken not setting status for gpx).

File: app/src/main/org/runnerup/export/Synchronizer.java [hunk 3]
```diff
@@ -70,8 +70,8 @@ public static int getInt(ExternalIdStatus s) {
         LIVE, // live feed of activity
         SKIP_MAP, // skip map in upload
         ACTIVITY_LIST, //list recorded activities
-        GET_ACTIVITY //download recorded activity
-
+        GET_ACTIVITY, //download recorded activity
+        FILE_FORMAT // upload as file in different possible formats
```
Comment (gerhardol) [inline app/src/main/org/runnerup/export/Synchronizer.java]: To set this as a feature works for two formats. Adding a third or variants (like dynamic setting of simplification) will be difficult. Of course, this field can be renamed to FILE_FORMAT_GPX_TCX. Is there another way?

File: app/src/main/org/runnerup/view/DetailActivity.java [hunk 4]
```diff
@@ -1,964 +1,965 @@
...
+        {
+            /*
+             * Accounts/reports
+             */
+            String sql = "SELECT DISTINCT "
+                    + "  acc._id, "
+                    + ("  acc." + DB.ACCOUNT.NAME + ", ")
+                    + ("  acc." + DB.ACCOUNT.FLAGS + ", ")
+                    + ("  acc." + DB.ACCOUNT.AUTH_CONFIG + ", ")
+                    + ("  acc." + DB.ACCOUNT.FORMAT + ", ")
```
Comment (gerhardol) [inline app/src/main/org/runnerup/view/DetailActivity.java]: why addformat here?

**Comment by gerhardol** [comment]
Can you please rebase?
There are some open questions, but I will probably merge as is.

**Comment by opoto** [comment]
Done

File: common/src/main/res/values-fr/strings.xml [hunk 5]
```diff
@@ -350,4 +350,6 @@
   <string name="path_simplification_algorithm">Algorithme</string>
   <string name="path_simplification_algorithm_info">Choisissez entre rapidité et haute qualité de la simplification de la trace</string>
   <string name="path_simplification_menu">Simplifier la trace</string>
+  <string name="File_format">Fichier(s) au format:</string>
```
Comment (gerhardol) [inline common/src/main/res/values-fr/strings.xml]: Need to be added to Transifex too, but I will handle this separately (normally Transifex will overwrite)

**Comment by opoto** [comment]
Rebased

**Comment by gerhardol** [comment]
Thanks!

**Pull request merged**